### ISSUE TITLE
Custom Directives not working

### ISSUE THREAD

**Issue by mogupta-ecfmg** [dotansimha/graphql-code-generator#7283]
**Describe the bug**
I am trying to create custom directives in a node.js/typescript project that uses graphql-codegen approach to build the schema. However, the directive transformer function is not getting executed when query is made.

**To Reproduce**
I referred to the [upperDirective](https://www.graphql-tools.com/docs/schema-directives#uppercasing-strings) sample to get started with writing a very simple code where entire schema is in one single file. Everything worked fine - the directive transformer function was executed when 'hello' query was called.

Then, I moved on to the codegen approach. And, the directive transformer function is NOT getting executed anymore.

[Github](https://github.com/mogupta-ecfmg/codegen-directives.git) link for the source code

Path to relevant files:

- starting point (azure http function)= sampleHTTPTriggerFunction\index.ts
- codegen config file = codegen.yml

**Expected behavior**
When following query is run

```graphql
query  {
  hello
}
```

then the expected output is 
```json
{
  "data": {
    "hello": "HELLO WORLD!"
  }
}
````

but codegen approach is returning following output:
```json
{
  "data": {
    "hello": "Hello World!"
  }
}
```

**Environment:**

**NodeJS: package.json:**
```json
{
  "name": "",
  "version": "",
  "scripts": {
    "build": "tsc",
    "build:production": "npm run prestart && npm prune --production",
    "watch": "tsc --w",
    "prestart": "npm run build && func extensions install",
    "start:host": "func start",
    "start": "npm-run-all --parallel start:host watch",
    "test": "jest",
    "gen": "graphql-codegen --config codegen.yml",
    "gen:watch": "graphql-codegen --config codegen.yml --watch"
  },
  "description": "",
  "devDependencies": {
    "@azure/functions": "^1.0.1-beta1",
    "@graphql-codegen/cli": "^2.3.0",
    "@graphql-codegen/introspection": "^2.1.0",
    "@graphql-codegen/typed-document-node": "^2.2.1",
    "@graphql-codegen/typescript": "^2.4.1",
    "@graphql-codegen/typescript-operations": "^2.2.1",
    "@graphql-codegen/typescript-resolvers": "^2.4.2",
    "@types/jest": "^26.0.24",
    "jest": "^27.0.6",
    "npm-run-all": "^4.1.5",
    "ts-jest": "^27.0.4",
    "typescript": "^4.5.4"
  },
  "dependencies": {
    "@graphql-tools/graphql-file-loader": "^7.3.3",
    "@graphql-tools/load": "^7.5.0",
    "@graphql-tools/load-files": "^6.5.3",
    "@graphql-tools/schema": "^8.3.1",
    "@graphql-tools/stitch": "^8.4.3",
    "@graphql-tools/utils": "^8.5.5",
    "apollo-server": "^3.5.0",
    "apollo-server-azure-functions": "^3.5.0",
    "graphql": "^16.2.0",
    "graphql-scalars": "^1.14.1",
    "underscore": "^1.13.2"
  }
}
```

codegen utility can be executed by running following command in terminal:
`npm run gen`

**Codegen Config:**
```yml
overwrite: true
schema: 
  - "./sampleHTTPTriggerFunction/resolvers/**/**.graphql"
  - "./sampleHTTPTriggerFunction/typedefs/**/**.graphql"
generates:
  sampleHTTPTriggerFunction/generated.ts:
    config:
      contextType: "./context#Context"
      useIndexSignature: true
    plugins:
      - "typescript"
      - "typescript-resolvers"
  ./graphql.schema.json:
    plugins:
      - "introspection"
hooks:
  afterAllFileWrite:
    - npx prettier --write
 ```

**Comment by dotansimha** [comment]
Hi,
I think we are missing something here - the directives are part of `graphql-tools` and they affect only the runtime of the server. 
Codegen doesn't have anything to do with the execution or manipulation done for the resolvers.

**Comment by mogupta-ecfmg** [comment]
Thanks. I did some more investigation on this issue. You are correct - this issue is not related to codegen. 

I initially thought it was related to codegen because it got introduced in our app when we introduced codegen approach. But there were other libraries also which were added to the project in order to build the schema from multiple .graphql files - and that's where the issue is. Apparently, loadSchemaSync function in @graphql-tools/load library does not pick up directives when merging schemas. I found out that there is another function loadTypedefsSync in the same library that works well with the directives.

Below is the sample code for anyone who is interested:

```
const sources = loadTypedefsSync(
        "./typedefs/**/*.graphql",
        {
        loaders: [new GraphQLFileLoader()],
      });
const documentNodes = sources.map(source => source.document);
const typeDefs = gql `${documentNodes[0]}`
const schema = makeExecutableSchema({
  typeDefs:typeDefs
})
```

Here is the [SO link](https://stackoverflow.com/questions/62441400/trouble-migrating-from-graphql-import-to-just-graphql-tools-with-apolloserver-d) which helped with this.
